app/controllers/cloud_object_store_container_controller.rb

```diff
@@ -19,10 +19,9 @@ def button
 
     process_cloud_object_storage_buttons(params[:pressed])
 
-    if !@flash_array.nil? && params[:pressed].ends_with?("delete")
-      javascript_redirect :action      => 'show_list',
-                          :flash_msg   => @flash_array[0][:message],
-                          :flash_error => @flash_array[0][:level] == :error
+    if !@flash_array.nil? && params[:pressed].ends_with?("delete") && @display != "cloud_object_store_objects"
+      session[:flash_msgs] = @flash_array.dup
+      javascript_redirect previous_breadcrumb_url
```

Comment:
You are right, I don't know how this could have slipped through my testing. Fixed.